### PR TITLE
Permission

### PR DESCRIPTION
**Pull request by facucompe**
# Sumary
* User model changed
* It creates admin from an existing and from an unexisting user
* It checks the creator is autentificated
# Trello card 
https://trello.com/c/aDfTTMgr/18-usuario-administrador
# Screenshot
image
image

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -51,6 +55,40 @@ exports.singIn = (req, res, next) => {
...
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    password: req.body.password,
+    email: req.body.email
+  } : {};
+
+  const accessToken = sessionManager.decode(req.headers.authorization);
+
+  userService.getByID(accessToken.id).then((us) => {
+    if (us.verificationCode === accessToken.verificationCode && moment().isBefore(accessToken.maximumUsefulDate)) {
```
Comment (mishuagopian) [inline app/controllers/users.js]: Use middleware for this..

File: app/controllers/users.js [hunk 2]
```diff
@@ -51,6 +55,40 @@ exports.singIn = (req, res, next) => {
...
+      userService.get(user).then((us2) => {
```
Comment (mishuagopian) [inline app/controllers/users.js]: return this

Comment (mishuagopian) [inline app/controllers/users.js]: You should get this user by email and not all the properties

File: app/controllers/users.js [hunk 3]
```diff
@@ -51,6 +55,40 @@ exports.singIn = (req, res, next) => {
...
+        if (us2) {
+          const valid = bcrypt.compareSync(user.password, us2.password);
```
Comment (mishuagopian) [inline app/controllers/users.js]: Hmm, no need of checking this. It's no sense..

File: app/controllers/users.js [hunk 4]
```diff
@@ -51,6 +55,40 @@ exports.singIn = (req, res, next) => {
...
+          if (valid) {
```
Comment (mishuagopian) [inline app/controllers/users.js]: This `if` will be removed

File: app/controllers/users.js [hunk 5]
```diff
@@ -51,6 +55,40 @@ exports.singIn = (req, res, next) => {
...
+            res.end();
+          } else {
+            next(errors.invalidUser);
+          }
+        } else {
+          newUser(req, res, next, user, 'admin');
+        }
+      }).catch(next);
+    } else {
+      next(errors.inexistingUser);
```
Comment (mishuagopian) [inline app/controllers/users.js]: Change `inexistingUser` with `notFound`

File: test/testUsers.js [hunk 6]
```diff
@@ -127,3 +127,50 @@ describe('/users GET', () => {
...
+      .set('authorization', accessToken)
+      .send({ firstName: 'firstName1', lastName: 'lastName1', username: 'username1',
+        password: 'password123', email: 'email1@wolox.com.ar' })
+      .then((res2) => {
+        res2.should.have.status(200);
+      });
+    })
+      .then(() => done());
+  });
+  it('Should return correct status after an update', (done) => {
```
Comment (mishuagopian) [inline test/testUsers.js]: Why do we have the exact same test here?

Comment (facucompe) [inline test/testUsers.js]: The first one creates a new user, the second one updates to admin an existing one

**Comment by mishuagopian** [comment]
@facucompe this PR should be merged once you finish with `usersList` and `autentication` branches

**Pull request merged**